pkg/workflows/management/upgrade_cluster.go

```diff
@@ -0,0 +1,72 @@
+package management
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/eks-anywhere/pkg/logger"
+	"github.com/aws/eks-anywhere/pkg/task"
+	"github.com/aws/eks-anywhere/pkg/workflows"
+)
+
+type upgradeCluster struct{}
+
+// Run upgradeCluster performs actions needed to upgrade the management cluster.
+func (s *upgradeCluster) Run(ctx context.Context, commandContext *task.CommandContext) task.Task {
+	// TODO(g-gaston): move this to eks-a installer and eks-d installer
+	err := commandContext.EksdInstaller.InstallEksdManifest(ctx, commandContext.ClusterSpec, commandContext.ManagementCluster)
+	if err != nil {
+		commandContext.SetError(err)
+		return &workflows.CollectMgmtClusterDiagnosticsTask{}
+	}
+
+	if commandContext.UpgradeChangeDiff.Changed() {
+		if err := commandContext.ClusterManager.ApplyBundles(ctx, commandContext.ClusterSpec, commandContext.ManagementCluster); err != nil {
+			commandContext.SetError(err)
+			return &workflows.CollectMgmtClusterDiagnosticsTask{}
+		}
+
+		if err := commandContext.ClusterManager.ApplyReleases(ctx, commandContext.ClusterSpec, commandContext.ManagementCluster); err != nil {
+			commandContext.SetError(err)
+			return &workflows.CollectMgmtClusterDiagnosticsTask{}
+		}
+	}
```

Comment:
Could we move this to a different task as well?
What do you think, I see this different than an "upgrade"
Is more like "install new eks-a version components"

Reply:
I think we should keep this under the upgrade process because to me it appears as a part of upgrade process. I am happy to move this around though.

Reply:
Interesting
Why do think it's part of the upgrade step?
These two operations don't make any changes to the cluster or affect any component running

Reply:
Yes, I got your point. I have moved this to a different task.